### PR TITLE
add null checks when adding to jeeComponentClass

### PR DESCRIPTION
**Pull request by benjamin-confino**

### CODE REVIEW

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java [hunk 1]
```diff
@@ -369,6 +377,9 @@ private void initializeJEEComponentClasses(Set<String> allClassNames) throws CDI
         for (EjbDescriptor<?> ejb : ejbs) {
             if (ejb.isMessageDriven()) {
                 classes.add(ejb.getBeanClass());
+                if (ejb.getBeanClass() == null) {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java]: For this case (and the one for managed beans) do we want to add the class to `classes` if it's `null`?

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java [hunk 2]
```diff
@@ -786,6 +797,9 @@ public void addEjbDescriptor(EjbDescriptor<?> ejbDescriptor) {
     public void addManagedBeanDescriptor(ManagedBeanDescriptor<?> managedBeanDescriptor) {
         if (getBeanDiscoveryMode() != BeanDiscoveryMode.NONE) {
             this.managedBeanClasses.add(managedBeanDescriptor.getBeanClass());
+            if (managedBeanDescriptor.getBeanClass() -= null) {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java]: `-=` is probably not what you meant here

Comment (benjamin-confino) [reply dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java]: Correct, that's a typo

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java (hunk 1 above)
Comment (benjamin-confino) [reply dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java]: As I said in slack. I think for now we stick to the existing behaviour, but log it.

**Pull request merged**